tests/Unit/NumericalAlgorithms/LinearSolver/LinearSolverAlgorithmTestHelpers.hpp

```diff
@@ -175,6 +176,49 @@ struct ElementArray {
   }
 };
 
+struct CleanOutput {
```

Comment:
It would be useful if you mentioned in the commit message what the changes are about. It was not obvious from the message or adding the observe part that this phase would be added. A comment above the `CleanOutput` struct describing why it's needed would also be very helpful (I understand why, I just think it should be documented).

Reply:
I like the updated commit message :) Could you also add a C++ comment here saying something like
```cpp
// After the algorithm completes we perform a cleanup phase that checks the expected output file was written and deletes it.
```